### DISCUSSION TITLE
Pattern for stable empty results

### DISCUSSION THREAD

**Discussion by pke** [vercel/swr#1034] [Q&A]
I wonder what the best pattern for stable "empty" results is, especially in regards to minimised re-renderings

## Variant #1 (Anti-Pattern?)

using a dynamic empty object as result

```ts
function myHook() {
  const { data } = useSWR(..)
  return {
    items: data || [] // non-stable
  }
}
```

## Variant #2

using a stable empty object

```ts
const empty = []
function myHook() {
  const { data } = useSWR(..)
  return {
    items: data || empty // stable
  }
}
```

## Variant #3 

using a default assignment

```ts
function myHook() {
  const { data = [] } = useSWR(..)
  return {
    items: data // stable?
  }
}
```

Variant clearly creates a new result on every hook rerender, which I would consider bad. Which patterns do you recommend?

**Comment by huozhi** [comment]
`initialData` is sth you're looking for. https://swr.vercel.app/docs/options

**Comment by ErnestBrandi** [comment]
EDIT : not so relevant after all.
Just be aware that data is `undefined` if nothing has been fetched yet and `null` if something went wrong white fetching.
Approach 1) is fine, you might want to use [??](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Nullish_coalescing_operator) instead of `||` though.
